### ISSUE TITLE
NuGet 3 and 4 binaries can't publish modules with 2 or 4 numbers in the version

### ISSUE THREAD

**Issue by Jaykul** [PowerShell/PowerShellGetv2#131]
Given a module 'Autoload' with a ModuleVersion = "4.1"
And you have NuGet 4 available in your Path.
When you `Publish-Module -Name Autoload` 
Then PowerShellGet fails to publish with an unhelpful error

In fact, the error was extremely difficult to troubleshoot -- I ended up having to set a breakpoint in the PowerShellGet module itself to figure out what it meant.

## Expected Behavior

Publish-Module should work.

## Current Behavior

```
NuGet.exe : File does not exist (C:\Users\Jaykul\AppData\Local\Temp\478508696\Autoload\Autoload.4.1.nupkg).
```

The reason is that you've created a file `Autoload.4.1.0.nupkg` but passed the wrong path to `nuget push`

**Comment by bmanikm** [comment]
@Jaykul From the error details, it looks like you are using 3.0 or later version of NuGet.exe.
This issue should not happen with https://www.NuGet.org/NuGet.exe .

It looks like, we need to parse the output of `NuGet.exe help` to find the NuGet.exe version and change the nupkg filename accordingly in the path for `nuget push`. Any other alternatives?


#### Workaround for others until this issue gets resolved in PowerShellGet
Please bootstrap the NuGet.exe from https://www.NuGet.org/NuGet.exe.

**Comment by Jaykul** [comment]
Ok, I've tested and you're exactly right.  It works fine with NuGet 2.x but not with Nuget 3 or Nuget 4 -- you guys are _**really far behind the times**_.

The issue was that we were trying to get publishing working with the VisualStudio.com galleries, and _incorrectly_ thought that we needed to use the _nuget.exe_ binary they provided in their authentication package -- it turns out it's enough to just use the _CredentialProvider_ with the version of NuGet that PowerShellGet uses -- even though [the documentation](https://docs.microsoft.com/en-us/nuget/api/nuget-exe-credential-providers) says you need NuGet 3.3 or higher.

**Comment by bmanikm** [comment]
This has been resolved with PR #196.
